adaptive_breakpoints/lib/src/adaptive_breakpoints.dart

```diff
@@ -0,0 +1,300 @@
+    portrait: 'medium handset',
+    window: AdaptiveWindowType.xs,
+    columns: 4,
+    margins: 16.0,
+    gutters: 16.0,
+  ),
+  BreakpointSystemEntry(
+    range: RangeValues(400, 479),
+    portrait: 'large handset',
+    window: AdaptiveWindowType.xs,
+    columns: 4,
+    margins: 16.0,
+    gutters: 16.0,
+  ),
+  BreakpointSystemEntry(
+    range: RangeValues(480, 599),
+    portrait: 'large handset',
+    landscape: 'small handset',
+    window: AdaptiveWindowType.xs,
+    columns: 4,
+    margins: 16.0,
+    gutters: 16.0,
+  ),
+  BreakpointSystemEntry(
+    range: RangeValues(600, 719),
+    portrait: 'small tablet',
+    landscape: 'medium handset',
+    window: AdaptiveWindowType.s,
+    columns: 8,
+    margins: 16.0,
+    gutters: 16.0,
+  ),
+  BreakpointSystemEntry(
+    range: RangeValues(720, 839),
+    portrait: 'large tablet',
+    landscape: 'large handset',
+    window: AdaptiveWindowType.s,
+    columns: 8,
+    margins: 24.0,
+    gutters: 24.0,
+  ),
+  BreakpointSystemEntry(
+    range: RangeValues(840, 959),
+    portrait: 'large tablet',
+    landscape: 'large handset',
+    window: AdaptiveWindowType.s,
+    columns: 12,
+    margins: 24.0,
+    gutters: 24.0,
+  ),
+  BreakpointSystemEntry(
+    range: RangeValues(960, 1023),
+    landscape: 'small tablet',
+    window: AdaptiveWindowType.s,
+    columns: 12,
+    margins: 24.0,
+    gutters: 24.0,
+  ),
+  BreakpointSystemEntry(
+    range: RangeValues(1024, 1279),
+    landscape: 'large tablet',
+    window: AdaptiveWindowType.m,
+    columns: 12,
+    margins: 24.0,
+    gutters: 24.0,
+  ),
+  BreakpointSystemEntry(
+    range: RangeValues(1280, 1439),
+    landscape: 'large tablet',
+    window: AdaptiveWindowType.m,
+    columns: 12,
+    margins: 24.0,
+    gutters: 24.0,
+  ),
+  BreakpointSystemEntry(
+    range: RangeValues(1440, 1599),
+    portrait: 'small handset',
+    window: AdaptiveWindowType.l,
+    columns: 12,
+    margins: 24.0,
+    gutters: 24.0,
+  ),
+  BreakpointSystemEntry(
+    range: RangeValues(1600, 1919),
+    portrait: 'small handset',
+    window: AdaptiveWindowType.l,
+    columns: 12,
+    margins: 24.0,
+    gutters: 24.0,
+  ),
+  BreakpointSystemEntry(
+    range: RangeValues(1920, double.infinity),
+    portrait: 'small handset',
+    window: AdaptiveWindowType.xl,
+    columns: 12,
+    margins: 24.0,
+    gutters: 24.0,
+  ),
+];
+
+/// Returns the [AdaptiveWindowType] type to the user.
+///
+/// This is useful when the user wants to compare the MediaQuery to the current
+/// window size.
+AdaptiveWindowType getWindowSize(BuildContext context) {
+  return getBreakpointEntry(context).window;
+}
+
+BreakpointSystemEntry getBreakpointEntry(BuildContext context) {
+  double width = MediaQuery.of(context).size.width;
+  for (BreakpointSystemEntry entry in breakpointSystem) {
+    if (entry.range.start <= width && width < entry.range.end + 1) {
+      return entry;
+    }
+  }
+  throw AssertionError('Size is too large');
```

Comment:
This error message needs to be improved. Does this means a user is using a device larger than xl? What should they do if they encounter this?

Reply:
Done